### DISCUSSION TITLE
Multiple Serial Port Configuration for Modbus

### DISCUSSION THREAD

**Discussion by Hambren** [artisan-roaster-scope/artisan#1770] [Q&A]
I need to be able to configure Modbus56, 78 etc on different Com ports for different devices. I have got Modbus and Modbus34 configured fine with both using the same port and can easily add  56, 78 etc to that - but cannot for the life of me figure out how to get the necessary different port information that I need configured into Extra Devices for Modbus56 78 etc. I would much appreciate any guidance anyone has!

**Comment by MAKOMO** [comment]
This is simply not possible.

**Comment by Hambren** [comment]
Thanks Marco - I got to suspecting that this might be the case with Artisan which is a pity as many modbus devices (such as brushless motor controllers etc!) have their individual port configurations hard-coded and thus cannot be used in combination directly with Artisan software. However for others who have the same issue as me it looks like there is fortunately a standalone hardware solution available in the form of a milti-port serial server which can take master commands from the Artisan interface and do the port conversion gymnastics to serve up to four different slave outlet configurations  - see https://www.zlmcu.com/en/products_ZLAN5443D.htm as an example. I will be obtaining one of these and will let you know how it goes as I am sure I will not be the only person wrestling with this issue!! Take care and thanks as always for the brilliant product!!
